### PR TITLE
Change pm2ps_parser to power_models_data

### PR DESCRIPTION
**Pull request by daniel-thom**
This will fix issue #433.  It changes PowerModels parsing to follow the model of PowerSystemTableData.

This model has one limitation:  it requires two steps from the user.
```
pm_data = PowerModelsData(file)
sys = System(pm_data)
```

There is alternate solution that we could consider for both PowerModelsData and PowerSystemTableData. What do you think of this?

```
function System(::Type{PowerModelsData}, file::Union{String, IO}; kwargs...)
function System(
    ::Type{PowerSystemTableData},
    directory::AbstractString,
    base_power::Float64,
    user_descriptor_file::AbstractString;
    kwargs...
)
```

### CODE REVIEW

File: src/parsers/standardfiles_parser.jl [hunk 1]
```diff
@@ -14,18 +14,6 @@ sys = parse_standard_files(
 
 """
 function parse_standard_files(file::String; kwargs...)
```
Comment (daniel-thom) [inline src/parsers/standardfiles_parser.jl]: I'm not sure that this function is helpful.  It provides a second way of doing the same thing but doesn't add value (other than being a single step).

Comment (jd-lara) [reply src/parsers/standardfiles_parser.jl]: We can remove this function and leave Systems as the only entry point for Matpower data.

**Comment by jd-lara** [comment]
> This will fix issue #433. It changes PowerModels parsing to follow the model of PowerSystemTableData.
> 
> This model has one limitation: it requires two steps from the user.
> 
> ```
> pm_data = PowerModelsData(file)
> sys = System(pm_data)
> ```
> 
> There is alternate solution that we could consider for both PowerModelsData and PowerSystemTableData. What do you think of this?
> 
> ```
> function System(::Type{PowerModelsData}, file::Union{String, IO}; kwargs...)
> function System(
>     ::Type{PowerSystemTableData},
>     directory::AbstractString,
>     base_power::Float64,
>     user_descriptor_file::AbstractString;
>     kwargs...
> )
> ```

I think that the current design is ok. I don't see too much value on the dispatch on a type. If you want to add it is ok but it seems that calling directly the file and detecting the extension is more straightforward.

**Comment by claytonpbarrows** [comment]
I like the single line `System` functionality. I'd be fine with the suggestion from @jd-lara to detect file extension to enable this. I'd also be fine with requiring users pass in the type. However, if we go with the latter, I'd suggest adding the following to help avoid confusion about what is PowerModelsData:
```julia
abstract type MATPOWER <: PowerModelsData
abstract type RAW <: PowerModelsData
```

**Comment by daniel-thom** [comment]
I changed my mind on my proposal above.  The PowerSystemTableData and System constructors are two complicated to combine. I think it's best to leave them independent.

**Pull request merged**